tests/unit_tests/test_analytics/sentry_relay.py

```diff
@@ -0,0 +1,161 @@
+from __future__ import annotations
+
+import gzip
+import socket
+import threading
+import time
+from typing import Any
+
+import flask
+import requests
+from flask import request
+from sentry_sdk.envelope import Envelope
+
+
+class SentryResponse:
+    def __init__(
+        self,
+        message: str | None,
+        project_id: str,
+        public_key: str,
+        tags: None | dict[str, Any] = None,
+        is_error: bool = False,
+    ):
+        self.message = message
+        self.project_id = project_id
+        self.public_key = public_key
+        self.tags = tags
+        self.is_error = is_error
+
+    def __eq__(self, other):
+        return (
+            self.message == other.message
+            and self.project_id == other.project_id
+            and self.public_key == other.public_key
+            and self.tags == other.tags
+            and self.is_error == other.is_error
+        )
+
+
+class SentryExceptionEventResponse(SentryResponse):
+    def __init__(
+        self,
+        message: str | None,
+        project_id: str,
+        public_key: str,
+        tags: dict[str, Any],
+        stacktrace: dict[str, Any],
+    ):
+        super().__init__(message, project_id, public_key, tags, is_error=True)
+        self.stacktrace = stacktrace
```

Comment:
why do you need this special class for adding the stack trace? it seems that it is enough to have the original one and just not set the stack trace argument

Reply:
My thoughts were that it feels like its starting to overload the SentryResponse class to much.
By splitting into the exception class the purpose of the class is more clear and it would allow more flexibility if additional methods, fields, or verification methods were needed later.

But we can do this later if needed.